src/core/crypto/crypto_platform.cpp

```diff
@@ -267,67 +306,98 @@ OT_TOOL_WEAK otError otPlatCryptoHkdfExtract(void *             aContext,
     return error;
 }
 
+otError otPlatCryptoHkdfDeinit(otCryptoContext *aContext)
+{
+    Error             error = kErrorNone;
+    HmacSha256::Hash *prk;
+
+    VerifyOrExit(aContext != nullptr, error = kErrorInvalidArgs);
+    VerifyOrExit(aContext->mContextSize >= sizeof(HmacSha256::Hash), error = kErrorFailed);
+
+    prk = static_cast<HmacSha256::Hash *>(aContext->mContext);
+    prk->~Hash();
```

Comment:
This is ok but we may be fine to not call destructor (since there is none).

The `Sha256::Hash` is really simple type (just a buffer to conatin the hash) and it is a mirror or a public OT definition (which is a C struct).
```c++
OT_TOOL_PACKED_BEGIN
struct otCryptoSha256Hash
{
    uint8_t m8[OT_CRYPTO_SHA256_HASH_SIZE]; ///< Hash bytes.
} OT_TOOL_PACKED_END;
```

Reply:
Added in case the class gets more complex in the future.